File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> storage;
        for(int i = 0; i < tokens.size(); i++){
            if((tokens[i] != "+") && (tokens[i] != "-") && (tokens[i] != "/") && (tokens[i] != "*")){
                storage.push(stoi(tokens[i]));
            }
            else{
                int a = storage.top();
                storage.pop();
                int b = storage.top();
                storage.pop();
                if(tokens[i] == "+"){
                    storage.push(b + a);
                }
                else if(tokens[i] == "-"){
                    storage.push(b - a);
                }
                else if(tokens[i] == "/"){
                    storage.push(b / a);
                }
                else if(tokens[i] == "*"){
                    storage.push(b * a);
                }
            }
        }

        return storage.top();
    }
};

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:

    def countArea(self, bar: Tuple[int, int], index: int) -> int:
        return bar[1] * (index - bar[0])

    def largestRectangleArea(self, heights: List[int]) -> int:

        stack = []
        maxArea = 0
        for index, height in enumerate(heights):
            top = ()
            while stack and stack[-1][1] > height:

                top = stack.pop()
                maxArea = max(
                    maxArea,
                    self.countArea(top, index)
                )
                
            if stack:
                if top:
                    stack.append((top[0], height))
                else:
                    stack.append((index, height))
                    
            else:
                stack.append((0, height))

        while stack:
            top = stack.pop()
            maxArea = max(
                    maxArea,
                    self.countArea(top, len(heights))
            ) 

        return maxArea